drivers/soundwire/bus.c

```diff
@@ -1331,7 +1333,7 @@ static int sdw_handle_dp0_interrupt(struct sdw_slave *slave, u8 *slave_status)
 		count++;
 
 		/* we can get alerts while processing so keep retrying */
-	} while (status != 0 && count < SDW_READ_INTR_CLEAR_RETRY);
+	} while ((status & SDW_DP0_INTERRUPTS) && (count < SDW_READ_INTR_CLEAR_RETRY));
```

Comment:
actually there's no need to add those parentheses around the `<` comparison. I find it intuitive enough that comparisons like `if (a < b && c < d)` don't need any more bracketing. The first pair of parentheses isn't needed either, but that indeed is less intuitive :-)

Reply:
I don't like having to think about operator precedence between a & b && c < d, and if we use parenthesis on one part of the formula we do in the second as well. no change.